mesa/space.py

```diff
@@ -459,14 +459,19 @@ def move_agent_to_one_of(
             elif selection == "closest":
                 current_pos = agent.pos
                 # Find the closest position without sorting all positions
-                closest_pos = None
+                closest_pos = []
                 min_distance = float("inf")
                 for p in pos:
```

Comment:
What if instead of keeping a list, we shuffle the `pos` list here? I think that's both faster and cleaner in implementation.

Reply:
Shuffling the list is a pretty simple solution so I don't mind doing that.

I printed all the pos lists for my simulation and most were pretty big but had a few closest locations. 

It would be faster to shuffle the shorter list like I did. However my idea needs to go over all the points which is slower.

But in terms of complexity the bulk of the work (complexity) is finding the smallest distance. Both ideas are the same. 

Before I found this method I was thinking of adding a distance representation masked layer ndarray so I could let numpy do this via a vectorized op. This would be cached by the agent after each move and reused.

But I did not explore this yet but it might be more efficient to build this for r =1 
check for a match, r=2 check.

Bottom line: I think to find a better answer would require bench-marking. It's probably not that important for this revision. Do as you see fit.